### DISCUSSION TITLE
How to add new column to users table?

### DISCUSSION THREAD

**Discussion by sk0le** [supabase/supabase#1334] [Questions]
Hello so I've setup pretty basic next app with supabase, something like this:  [LINK](https://codesandbox.io/s/elegant-hermann-fnmwm?file=/pages/index.js:1189-1193)

Now I need to add a custom column to the user table, something like this: 

![image](https://i.ibb.co/xhPjmHK/Screenshot-8.jpg)

As you can see the custom column that I need to add is a username, now how can I do it?

**Comment by gomflo** [comment]
You need to create a `public.users` table for extra columns.

https://supabase.io/docs/guides/auth#create-a-publicusers-table

**Reply by sk0le** [reply]
So I've added as you said, I have 2 tables and I've setup something like this:
```javascript

  const updateUser = async () => {
    return await supabase.from("users").update({ username }).eq("email", email);
  };

  const signUp = async () => {
    return await supabase.auth.signUp({
      email,
      password,
    });
  };

    const [userB, user] = await Promise.all([signUp(), updateUser()]);
    if (userB.error) setError(userB.error.message);
    else {
      if (user.error) setError(user.error.message);
      else {
        setMessage("Success! Now confirm your email!");
      }
    }
```

But I get error: https://prnt.sc/12cw3kb - now when I go to this route is says this: 
` Kong Error
No API key found in request.`

**Comment by sk0le** [comment]
I've changed the code to this and it's working.

The problem was when I call updateUser() there was no user to update.
This is code now:
```javascript
    const { data, error } = await supabase
      .from("users")
      .select()
      .match({ username });
    if (data) {
      if (data?.length < 1) {
        const userAuth = await signUp();
        if (userAuth.error) setError(userAuth.error.message);
        else {
          const userUpdated = await updateUser();
          if (userUpdated.error) setError(userUpdated.error.message);
          else {
            setMessage(
              "Successfully registered, please confirm your mail to proceed!"
            );
          }
        }
      } else {
        setError("Username already taken");
      }
    }
    if (error) {
      setError(error?.message);
    }
```
